File: 10684/main.cpp
#include <stdio.h>

int main(){
	int num, i;
	int cumtot=0, max=-1000000000, scanned;
	scanf("%d", &num);
	while(num!=0){
		for (i=0; i<num; i++){
			scanf("%d" ,&scanned);
			cumtot+=scanned;
			if(cumtot>max){max=cumtot;}
			if(cumtot<0){cumtot=0;}
		}
		if(max<=0){printf("Losing streak.\n");}
		else{printf("The maximum winning streak is %d.\n", max);}
		scanf("%d", &num);
		cumtot=0;max=-1000000000;	
	}

	return 0;
}

File: 10684/Main.java
import java.util.*;
import java.io.*;

public class Main{
	public static void main(String[] args){
		Scanner sc = new Scanner(System.in);
		int amount, cumtot, max;
		while(true){
			amount=sc.nextInt();
			if(amount==0){break;}
			cumtot=0;
			max=Integer.MIN_VALUE;
			for (int i=0; i<amount; i++){
				cumtot+=sc.nextInt();
				if(cumtot>max){
					max=cumtot;
				}
				if(cumtot<0){
					cumtot=0;
				}
			}
			if(max<=0){System.out.println("Losing streak.");}
			else{System.out.println("The maximum winning streak is "+max+".");}
		}
	}
}